lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -89,6 +70,27 @@ def fetch_logs
 
     private
 
+    def interpret_pod_status_data(status_data, metadata)
+      @status = @phase = (metadata["deletionTimestamp"] ? "Terminating" : status_data["phase"])
+
+      if @phase == "Failed" && status_data['reason'].present?
+        @status += " (Reason: #{status_data['reason']})"
+      elsif @phase != "Terminating"
+        ready_condition = status_data.fetch("conditions", []).find { |condition| condition["type"] == "Ready" }
+        @ready = ready_condition.present? && (ready_condition["status"] == "True")
+        @status += " (Ready: #{@ready})"
+      end
+    end
+
+    def log_suspicious_states(container_statuses)
+      container_statuses.each do |status|
+        waiting_state = status["state"]["waiting"] if status["state"]
+        reason = waiting_state["reason"] if waiting_state
```

Comment:
You could probably avoid those `if`s by using try:

```
reason = status.try!(:[], "state").try!(:[], "waiting").try!(:[], "reason")
```

Dunno if this chain makes the code better.